templates/new_service/.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,176 @@
+name: Build and Deploy
+concurrency: build_and_deploy_${{ github.ref_name }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: "Deploy environment"
+        required: true
+        type: choice
+        default: review
+        options:
+        - review
+        - development
+        - production
+      docker-image-tag:
+        description: "Docker image tag to deploy (optional)"
+        required: true
+        type: string
+      pull-request-number:
+        description: "Pull request number (required for review environment)"
+        required: false
+        type: string
+  push:
+   branches:
+    - main
+  pull_request:
+    branches:
+    - main
+    types: [opened, reopened, synchronize, labeled]
+
+env:
+  TERRAFORM_BASE: terraform/application
+  HEALTHCHECK_CMD: 'healthcheck/all'
+
+jobs:
+  build:
+    name: Build
+    if: ${{ github.event_name != 'workflow_dispatch' }}
+    env:
+      DOCKER_REPOSITORY: #DOCKER_REPOSITORY#
+    outputs:
+      docker-image-tag: ${{ steps.build-image.outputs.tag }}
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+      name: Checkout
+
+    - name: Build and push docker image
+      id: build-image
+      uses: DFE-Digital/github-actions/build-docker-image@master
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        context: .
+        docker-repository: ${{ env.DOCKER_REPOSITORY }}
+        max-cache: true
+        reuse-cache: true
+        snyk-token: ${{ secrets.SNYK_TOKEN }}
+
+  test:
+    name: Run tests
+    if: ${{ github.event_name != 'workflow_dispatch' }}
+    runs-on: ubuntu-latest
+
+    # This is a skeleton test job
+    # It will need to be updated to set the environment and run required tests
+    # The steps can be moved to a separate action if preferred
+
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v4
+
+    - name: Set up Ruby
```

Comment:
This may be used for non ruby apps like .NET app. I suggest to simply having 1 generic commented step like "Run tests"